File: readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>

/// index = 0.0588 * L - 0.296 * S - 15.8

int letters_counter(string s);
int words_counter(string s);
int sentences_counter(string s);
float grade(int letters, int words, int sentences);

int main(void)
{
    string s = get_string("Text: ");  
    int letters = letters_counter(s);
    int words = words_counter(s);
    int sentences = sentences_counter(s);
    
    float G = grade(letters, words, sentences);
    if (G > 16)
    {
        printf("Grade 16+\n");
    }
    else if (G < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %.0f\n", G);
    }
    
}


int letters_counter(string s)
{
    int letters = 0;
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        if (isalnum(s[i]) != 0) /// ceheck if alpahnumeric
        {
            letters += 1;
        }
    }
    return letters;
}

int words_counter(string s)
{
    int words = 0;
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        if (isblank(s[i]) != 0) /// ceheck if a blank, tab or so
        {
            words += 1;
        }
    }
    words += 1;
    return words;
}

int sentences_counter(string s)
{
    int sentences = 0;
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        if (s[i] == '!' || s[i] == '.' || s[i] == '?')  /// ceheck if a ! . or ?
        {
            sentences += 1;
        }
    }
    return sentences;
}



float grade(int letters, int words, int sentences)
{
    float L = (float)letters / (float)words*100;
    float S = (float)sentences / (float)words*100;
    float G = 0.0588 * L - 0.296 * S - 15.8;
    return G;
}

File: readability.py

# index = 0.0588 * L - 0.296 * S - 15.8

def main():
    string = input("Text: ")
    letters = letters_counter(string)
    words = len(string.split())
    sentences = sentences_counter(string)
    g = grade(letters, words, sentences)
    
    if g > 16:
        print("Grade 16+")
    elif g < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {round(g)}")
    
    
def letters_counter(str):
    letters = 0
    for l in str:
        if l.isalnum():
            letters += 1
    return letters
    
    
def sentences_counter(str):
    sentences = 0
    for s in str:
        if s == "!" or s == "." or s =="?":
            sentences += 1
    return sentences
    
    
def grade(letters, words, sentences):
    L = letters / words * 100
    S = sentences / words * 100
    G = 0.0588 * L - 0.296 * S - 15.8
    return G
    
    
main()